test/factories.rb

```diff
@@ -163,46 +162,33 @@
   end
 
   factory :user do
-    email { Faker::Internet.email }
+    sequence(:email) { |n| "john-#{n}@example.com" }
     password { "123456" }
     encrypted_password { Devise::Encryptor.digest(User, "123456") }
     first_name { Faker::Name.first_name }
     last_name { Faker::Name.last_name }
     tos_agreement { true }
 
-    organization
+    organization { ActsAsTenant.test_tenant }
 
     trait :verified_staff do
-      staff_account
+      staff_account { association :staff_account, organization: organization}
     end
 
     trait :staff_admin do
-      association(:staff_account, :admin)
+      staff_account { association :staff_account, :admin, organization: organization}
     end
 
     trait :unverified_staff do
-      staff_account { build(:staff_account, :unverified) }
+      staff_account { association :staff_account, :unverified, organization: organization}
     end
 
     trait :adopter_without_profile do
       adopter_account
     end
 
     trait :adopter_with_profile do
-      adopter_account
-
-      after :create do |user|
-        create :adopter_profile, adopter_account: user.adopter_account
-      end
-    end
-
-    trait :application_awaiting_review do
-      adopter_account
-
-      after :create do |user|
-        create :adopter_application, adopter_account: user.adopter_account, status: 0
-        user.adopter_account.adopter_profile = create(:adopter_profile)
-      end
```

Comment:
This trait was creating 4 different organizations under the hood. It was hard to fix so I decided it's better to remove it and build these associations in the test itself (test/integration/adoptable_pet_show_test.rb)